locale/CMakeLists.txt

```
@@ -22,9 +22,9 @@ foreach(_locale ${_locales})
     DEPENDS generate_mo_files_dir
     COMMENT "Create locale specific directories")
   add_custom_command(
-    OUTPUT generate_mo_files_dir_${_locale}
+    TARGET generate_mo_files_dir_${_locale}
```

```suggestion
    OUTPUT generate_mo_files_dir_${_locale}/LC_MESSAGES
```

Would this suggestion, together with the following change of L40:

```cmake
    DEPENDS generate_mo_files_dir_${_locale}/LC_MESSAGES
```

work?

(OUTPUT would in this case be preferable to TARGET).

---

> (OUTPUT would in this case be preferable to TARGET).

Could you elaborate more on this?

---

> ```cmake
> DEPENDS generate_mo_files_dir_${_locale}/LC_MESSAGES
> ```

The problem with `OUTPUT generate_mo_files_dir_${_locale}/LC_MESSAGES` is `generate_mo_files_dir_${_locale}/LC_MESSAGES` is not an actual output file or folder. We don't create `generate_mo_files_dir_${_locale}` folder. It should rather be something like `OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output/lib64/grass85/locale/${_locale}/LC_MESSAGES`, but I'm not sure if this is a better solution than `TARGET` unless we have a good reason for preferring `OUTPUT` over `TARGET`?

---

Ahh, you're right about it not being actual output file, in that case I'd suggest to use `${OUTDIR}/${GRASS_INSTALL_LOCALEDIR}/${_locale}/LC_MESSAGES` in all three places (L25, L27 and L40).

`OUTPUT` is better as it is a simple creation of a directory. If we were to use TARGET, then we also need to use `BYPRODUCTS` which in this case would sound contradictory.

---

Done!